Task 2: Adding instant search

sherinRoy opened the pull request.

kandarp-patel51 commented:
Add supported description and working application video or screenshots

kandarp-patel51 commented on libs/books/feature/src/lib/book-search/book-search.component.ts:
```diff
@@ -35,6 +39,24 @@ export class BookSearchComponent{
+    )
+    .subscribe(
+      () => this.searchBooks(),
+      (err) => console.error(err, 'Something went wrong')
```
We don't need console error, remove this line

kandarp-patel51 commented on libs/books/feature/src/lib/book-search/book-search.component.ts:
```diff
@@ -35,6 +39,24 @@ export class BookSearchComponent{
+
   searchExample() {
     this.searchForm.controls.term.setValue('javascript');
     this.searchBooks();
```
Do we need to call this method here?

kandarp-patel51 replied:
What I mean is do we need ```this.searchBooks()``` here?

kandarp-patel51 replied:
this.searchBooks(); - You don't need to call this method in searchExample() as you implemented instant search right?

kandarp-patel51 reviewed (changes requested):
Pls take a look at review comments

sherinRoy commented:
Adding instant search

- Added instant search functionality and removed existing search button.
- Added test cases to ensure instant search functionality works.

Screen shots

- Lint

- Unit test

- e2e test

- Task 2 Screenshot

sherinRoy commented on libs/books/feature/src/lib/book-search/book-search.component.spec.ts:
```diff
@@ -1,27 +1,98 @@
+      component.searchForm.controls.term.setValue('Javascript');
+      tick(500);
+
+      expect(store.dispatch).toHaveBeenCalledTimes(1);
```
test case is failing when changing this to 2

kandarp-patel51 replied:
Better way to expect is to compare the action parameter, for example:
```
 expect(store.dispatch).toHaveBeenCalledWith('Javascript');
```

if you do this, you can remove line no: 68 & 69

kandarp-patel51 commented on libs/books/feature/src/lib/book-search/book-search.component.spec.ts:
```diff
@@ -1,27 +1,96 @@
+      expect(store.dispatch).toHaveBeenCalledTimes(1);
+    }));
+
+    it('should not see the search results when received same input for search term', fakeAsync(() => {
```
I think, this should be changed to below one:

```
it('should see the existing search result when received same input for search term', fakeAsync(() => {
```

kandarp-patel51 reviewed (changes requested):
Pls take a look at review comment once

kandarp-patel51 commented on libs/books/feature/src/lib/book-search/book-search.component.html:
```diff
@@ -1,16 +1,22 @@
 <ng-container *ngIf="searchTerm; else empty">
   <div class="book-grid">
-    <div class="book" data-testing="book-item" *ngFor="let book of (books$|async)">
+    <div
```
Avoid such changes
